src/accounts/cosmos.ts

```diff
@@ -0,0 +1,74 @@
+import {
+    makeSignDoc,
+    OfflineAminoSigner,
+    Secp256k1HdWallet,
+    Secp256k1HdWalletOptions,
+    Secp256k1Wallet,
+} from "@cosmjs/amino";
+import { Account } from "./account";
+import { GetVerificationBuffer } from "../messages";
+import { BaseMessage, Chain } from "../messages/message";
+
+export class CosmosAccount extends Account {
+    private wallet: OfflineAminoSigner;
+
+    constructor(wallet: OfflineAminoSigner, address: string) {
+        super(address);
+        this.wallet = wallet;
+    }
+
+    GetChain(): Chain {
+        return Chain.CSDK;
+    }
+
+    async Sign(message: BaseMessage): Promise<string> {
+        const buffer = GetVerificationBuffer(message);
+
+        const aminoMsg = {
+            type: "signutil/MsgSignText",
+            value: {
+                message: buffer.toString(),
+                signer: message.sender,
+            },
+        };
+
+        const signDoc = makeSignDoc([aminoMsg], { amount: [], gas: "0" }, "signed-message-v1", "", "0", "0");
+        const { signature } = await this.wallet.signAmino(this.address, signDoc);
+
+        return JSON.stringify(signature);
+    }
+}
+
+async function getCosmosAccount(wallet: OfflineAminoSigner): Promise<CosmosAccount> {
+    const [account] = await wallet.getAccounts();
```

Comment:
is it possible to select which account to get? Right now, we are getting the first one, no?

Reply:
Or is the first one also the one the user currently selected through their wallet?